src/Sulu/Component/Content/Document/Subscriber/StructureRemoveSubscriber.php

```diff
@@ -57,19 +58,19 @@ public function handleRemove(RemoveEvent $event)
 
     public function removeDocument($document)
     {
-        // TODO: This is not a good indicator. There should be a RoutableBehavior here.
-        if (!$document instanceof StructureBehavior) {
-            return;
-        }
-
         if ($document instanceof ChildrenBehavior) {
             foreach ($document->getChildren() as $child) {
                 $this->removeDocument($child);
             }
         }
 
-        $this->removeReferences($document);
-        $this->recursivelyRemoveRoutes($document);
+        if ($document instanceof StructureBehavior) {
+            $this->removeReferences($document);
+        }
+
+        if ($document instanceof RoutableBehavior) {
```

Comment:
When thinking of it again... Wouldn't it be cleaner the current route would be deleted here (however we achieve that, the route could be manually deleted here and in the `CustomUrlSubscriber`), and it is the task of the `RouteSubscriber` to recursively remove the other routes?